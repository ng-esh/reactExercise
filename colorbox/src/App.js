import React from "react";
import BoxList from "./components/BoxList";
import "./styles/App.css";

function App() {
  return (
    <div className="app-container">
      <h1>Color Box Maker</h1>
      <BoxList />
    </div>
  );
}

export default App;
