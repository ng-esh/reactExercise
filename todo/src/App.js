import React from "react";
import TodoList from "./components/TodoList.js";
import "./App.scss";

function App() {
  return (
    <div className="app-container">
      <h1 data-testid="app-heading">Todo App</h1>
      <TodoList />
    </div>
  );
}

export default App;
