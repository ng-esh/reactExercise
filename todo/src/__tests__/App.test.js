import { render, screen } from '@testing-library/react';
import App from '../App.js';

test("renders Color Box Maker heading", () => {
  render(<App />);
  expect(screen.getByText("Todo App")).toBeInTheDocument();
});

test("renders without crashing", () => {
  render(<App />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
