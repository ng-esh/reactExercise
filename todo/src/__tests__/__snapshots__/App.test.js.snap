// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-container"
  >
    <h1
      data-testid="app-heading"
    >
      Todo App
    </h1>
    <div
      class="todo-list"
    >
      <form
        class="new-todo-form"
      >
        <input
          data-testid="todo-input"
          placeholder="Enter a new todo"
          type="text"
          value=""
        />
        <button
          data-testid="add-todo-btn"
          type="submit"
        >
          Add Todo
        </button>
      </form>
      <ul
        data-testid="todo-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
